packages/ui-extensions-dev-console/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,62 @@
+import styles from './Tooltip.module.css'
+import {TooltipPopover} from './TooltipPopover'
+import React, {useRef, useReducer} from 'react'
+import type {TooltipProps, TooltipState, TooltipAction} from './types'
+
+function tooltipReducer(state: TooltipState, action: TooltipAction) {
+  switch (action.type) {
+    case 'show':
+      return {...state, isVisible: true}
+    case 'hide':
+      return {...state, isVisible: false}
+    case 'position':
+      return {...state, position: action.payload}
+    default:
+      throw new Error()
+  }
+}
```

Comment:
It looks like there are just two actions here: 

1. Show, which sets isVisible and the position
2. Hide, which sets is visible

Is a reducer overkill?